http-server-netty/src/main/java/io/micronaut/http/server/netty/shortcircuit/DiscriminatorStage.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.netty.shortcircuit;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.MediaType;
+import io.micronaut.web.router.shortcircuit.MatchRule;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.QueryStringDecoder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The different stages of route matching. These are matched in order. For example the
+ * {@link #PATH} stage matches the request path to the candidate routes.<br>
+ * This class also handles transforming the {@link MatchRule}s corresponding to each stage to a
+ * {@link MatchPlan}.
+ *
+ * @author Jonas Konrad
+ * @since 4.3.0
+ */
+@Internal
+enum DiscriminatorStage {
+    PATH {
+        @Override
+        <R> MatchPlan<R> planDiscriminate(Map<MatchRule, MatchPlan<R>> nextPlans) {
+            // simplify already removed any overlaps between patterns and exact matches. only the exact matches are left
+            Map<String, MatchPlan<R>> byString = nextPlans.entrySet().stream()
+                .filter(e -> e.getKey() instanceof MatchRule.PathMatchExact)
+                .collect(Collectors.toMap(e -> ((MatchRule.PathMatchExact) e.getKey()).path(), Map.Entry::getValue));
+            return request -> {
+                // this replicates AbstractNettyHttpRequest.getPath but not exactly :(
```

Comment:
please clarify

Reply:
AbstractNettyHttpRequest.getPath for some reason uses the charset from the content-type of the request, if set. i do not want the charset parsing logic anymore. i dont think anyone uses it and it adds overhead to every request.